src/Microsoft.AspNet.JsonPatch/Converters/JsonPatchDocumentConverter.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNet.JsonPatch.Exceptions;
+using Microsoft.AspNet.JsonPatch.Operations;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json.Serialization;
+
+namespace Microsoft.AspNet.JsonPatch.Converters
+{
+    public class JsonPatchDocumentConverter : JsonConverter
+    {
+        public override bool CanConvert(Type objectType)
+        {
+            return true;
+        }
+
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, 
+            JsonSerializer serializer)
+        {
+            try
+            {
+                if (reader.TokenType == JsonToken.Null)
+                {
+                    return null;
+                }
+
+                if (objectType != typeof(JsonPatchDocument))
+                {
+                    throw new ArgumentException(Resources.ObjectTypeMustBeJsonPatchDocument, "objectType");
+                }
```

Comment:
This should be outside the `try`. As it is you'll catch and rethrow which isn't really helping.